modules/gateway/peers.go

```diff
@@ -604,3 +604,20 @@ func (g *Gateway) Peers() []modules.Peer {
 	}
 	return peers
 }
+
+// Online returns true if the node is connected to the internet. During testing
+// we always assume that the node is online
+func (g *Gateway) Online() bool {
+	if build.Release != "standard" {
+		return true
+	}
+
+	g.mu.RLock()
+	defer g.mu.RUnlock()
+	for _, p := range g.peers {
+		if !p.Local {
+			return true
+		}
+	}
```

Comment:
Eventually I'd like to see this switched to an atomic value that gets updated any time that we adjust the `g.peers` set, which means that we can call this function without having to do anything with locking.

Reply:
no need to do that this PR, but we should probably open an issue for it labeled `Programming Task` on the issue tracker.

Reply:
as I said in another comment, I don't think we need to worry about this code being highly performant. We aren't going to be calling `Online` very frequently, and the lock is held for a very short period. Adding more shared mutable state increases the likelihood of bugs and makes it harder for other devs to figure out how the value is calculated.

Reply:
An atomic variable would be overkill since we only update it when we update the peers for which we need a lock anyway.